File: 1753-maximum-score-from-removing-stones/1753-maximum-score-from-removing-stones.cpp
class Solution {
public:
    int maximumScore(int a, int b, int c) {
        priority_queue<int>pq;
        pq.push(a);
        pq.push(b);
        pq.push(c);
        int res = 0;
        while(!pq.empty()){
            auto a = pq.top();
            pq.pop();
            if(pq.empty()) return res;
            auto b = pq.top();
            pq.pop();
            res++;
            a--;
            b--;
            if(a>0)pq.push(a);
            if(b>0)pq.push(b);
        }
        return res;
    }
};

File: 1753-maximum-score-from-removing-stones/1753-maximum-score-from-removing-stones.java
class Solution {
    public int maximumScore(int a, int b, int c) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>((x, y) -> y - x);
        pq.add(a);
        pq.add(b);
        pq.add(c);
        int res = 0;
        while(pq.size()>1){
            int p = pq.remove();
            int q = pq.remove();
            res++;
            p--;
            q--;
            if(p>0) pq.add(p);
            if(q>0) pq.add(q);
        }
        return res;
    }
}